src/context/AuthModalContext.tsx

```diff
@@ -1,49 +1,110 @@
-import { FunctionComponent, ReactNode, createContext, useContext, useState } from 'react'
+import {
+  FunctionComponent,
+  ReactNode,
+  createContext,
+  useContext,
+  useState,
+  useMemo,
+  useCallback,
+} from 'react';
 
-import { AuthenticateModalContent, Modal } from '../components'
-import { logAuthPopoverEvent } from '../util'
+import { useRouter } from 'next/router';
+import { useFeatureFlagVariantKey } from 'posthog-js/react';
+
+import {
+  AuthenticateModalContent,
+  Modal,
+  IdeModalContent,
+} from '../components';
+import { EventName, getEventLogger } from '../hooks/eventLogger';
+import { logAuthPopoverEvent } from '../util';
 
 interface AuthModalContextProps {
-    isSignUpModalOpen: boolean
-    openModal: (source: string, plan?: 'free' | 'pro') => void
-    closeModal: () => void
+  isSignUpModalOpen: boolean;
+  openModal: (source: string, plan?: 'free' | 'pro') => void;
+  closeModal: () => void;
 }
 
 const AuthModalContext = createContext<AuthModalContextProps>({
-    isSignUpModalOpen: false,
-    openModal: () => {},
-    closeModal: () => {},
-})
-
-export const useAuthModal = (): AuthModalContextProps => useContext(AuthModalContext)
-
-export const AuthModalProvider: FunctionComponent<{ children: ReactNode }> = ({ children }) => {
-    const [isSignUpModalOpen, setIsSignUpModalOpen] = useState(false)
-    const [source, setSource] = useState<string>('')
-    const [plan, setPlan] = useState<'pro' | 'free' | undefined>('free')
-
-    const openModal = (source: string, plan?: 'pro' | 'free'): void => {
-        setSource(source)
-        setIsSignUpModalOpen(true)
-        logAuthPopoverEvent(source)
-        setPlan(plan ?? 'free')
-    }
+  isSignUpModalOpen: false,
+  openModal: () => {},
+  closeModal: () => {},
+});
+
+export const useAuthModal = (): AuthModalContextProps =>
+  useContext(AuthModalContext);
+
+export const AuthModalProvider: FunctionComponent<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const router = useRouter();
+  const [isSignUpModalOpen, setIsSignUpModalOpen] = useState(false);
+  const [isControlModalOpen, setIsControlModalOpen] = useState(false);
+
+  const [source, setSource] = useState<string>('');
+  const [plan, setPlan] = useState<'pro' | 'free' | undefined>('free');
+
+  const userFlag = useFeatureFlagVariantKey('install-first');
+
+  const logEvent = (eventArguments: { group: string; modal: string }): void => {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    getEventLogger().log(EventName.ENROLLMENT, eventArguments, eventArguments);
+  };
 
-    const closeModal = (): void => {
-        setIsSignUpModalOpen(false)
+  const displayModal = useCallback((): void => {
+    const eventArguments = {
```

Comment:
Do you plan to use the `EventName.ENROLLMENT` event name for future A/B tests as well? Or new event names for every test? If the former, please add some sort of test identifier here so we can differentiate one test from another.